zkevm-circuits/src/copy_circuit.rs

```diff
@@ -328,41 +322,33 @@ impl<F: Field> SubCircuitConfig<F> for CopyCircuitConfig<F> {
 
             cb.condition(is_first.expr(),
                 |cb| {
-                    cb.require_zero("reader word_index starts at 0", meta.query_advice(word_index, Rotation::cur()));
-                    cb.require_zero("writer word_index starts at 0", meta.query_advice(word_index, Rotation::next()));
+                    // Apply the same constraints on the first reader and first writer rows.
+                    for rot in [Rotation::cur(), Rotation::next()] {
+                        cb.require_zero("word_index starts at 0", meta.query_advice(word_index, rot));
+
+                        cb.require_equal(
+                            "value_acc init to the first value, or 0 if padded or masked",
+                            meta.query_advice(value_acc, rot),
+                            meta.query_advice(value, rot) * meta.query_advice(non_pad_non_mask, rot),
+                        );
+                    }
                 },
             );
 
-            cb.condition(
-                and::expr([
-                    is_continue.expr(),
-                    not_word_end.expr(),
-                ]),
+            cb.condition(is_continue.expr(),
                 |cb| {
-                    cb.require_equal(
-                        "word_index[0] + 1 == word_index[2]",
-                        meta.query_advice(word_index, Rotation::cur()) + 1.expr(),
-                        meta.query_advice(word_index, Rotation(2)),
-                    )
-                },
-            );
 
-            cb.condition(
-                and::expr([
-                    is_continue.expr(),
-                    is_word_end.expr(),
-                ]),
-                |cb| {
-                    cb.require_equal(
-                        "word_index[0] == 31",
-                        meta.query_advice(word_index, Rotation::cur()),
-                        31.expr(),
+                    let inc_or_reset = select::expr(
```

Comment:
This is refactored into a single block using `select`. This shows clearly that both cases are handled (word end or not).

Reply:
good refactoring for clean code.